### PR TITLE
ci: Add branch and remote to Bazel get_workspace_status

### PR DESCRIPTION
**Pull request by WillEngFlow**
Signed-off-by: Will Martin <will@engflow.com>

Commit Message:
This change adds the BUILD_SCM_REMOTE and BUILD_SCM_BRANCH variables to those reported by Bazel. This will allow downstream consumers of the BEP within the CI system to include information regarding the repository and branch that a build was triggered from.

Additional Description: 
Specifically, this adds repository and branch to the [EngFlow UI][1] for envoy-mobile but may be used by any BEP consumer.

Risk Level: Low
Testing: Manual
Docs Changes: None
Release Notes: None
Platform Specific Features: N/A

[1]: https://envoy.cluster.engflow.com/

### CODE REVIEW

**Review by phlax — DISMISSED** [review]
lgtm, thanks @WillEngFlow

**Comment by phlax** [comment]
seems like its failing shellcheck

```console
In bazel/get_workspace_status line 48:
if [[ $? == 0 ]]; then
      ^-- SC2181 (style): Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
```

**Comment by WillEngFlow** [comment]
Ah, thanks guys! This is a little tricky because I want to store the value in a variable and read the exit code separately. Let me try a different way and see if I can please shellcheck!

File: bazel/get_workspace_status
Comment (phlax) [inline bazel/get_workspace_status:48]: how about just checking if `git_remote` was set - shellcheck doesnt like checking the return code and i cant off hand think of a way of iffing and setting at the same time - however, i think if previous command exitted non 0 then `git_remote` would not be set - so i think you can `if [[ -n "$git_remote" ]]; then ...`

Comment (phlax) [reply bazel/get_workspace_status:48]: for more assurance you could change to...

```bash
git_remote=$(git remote get-url origin || echo "")
```

Comment (WillEngFlow) [reply bazel/get_workspace_status]: I like that idea, let me give it a go!

**Review by phlax — APPROVED** [review]
lgtm, thanks @WillEngFlow

**Comment by WillEngFlow** [comment]
Thanks for walking me through that @phlax! Is there any more action required on my part before this can be merged?

**Comment by phlax** [comment]
can be merged - i was waiting on ci

**Pull request merged**
